features/url-formattering.feature

```diff
@@ -0,0 +1,127 @@
+        | adres | https://api.bag.acceptatie.kadaster.nl/esd/huidigebevragingen/v1/adressen?adresseerbaarObjectIdentificatie=0518010000781379 |           |
+        En bevat de response de volgende HTTP header
+        | naam               | waarde        |
+        | Content-Uri-Format | uri-reference |
+
+Rule: 'uri-reference' is de default uri format
+
+    Scenario: er is geen Accept-Uri-Format header in de request
+        Als GET "/ingeschrevenpersonen?burgerservicenummer=999993847" wordt aangeroepen zonder HTTP header "Accept-Uri-Format"
+        Dan bevat _links in de response minimaal de volgende links
+        | naam  | href                                                                                                                        | templated |
+        | base  | https://www.haalcentraal.nl/haalcentraal/api/brp                                                                            |           |
+        | self  | /ingeschrevenpersonen?burgerservicenummer=999993847                                                                         |           |
+        | adres | https://api.bag.acceptatie.kadaster.nl/esd/huidigebevragingen/v1/adressen?adresseerbaarObjectIdentificatie=0518010000781379 |           |
+        En bevat de response de volgende HTTP header
+        | naam               | waarde        |
+        | Content-Uri-Format | uri-reference |
+
+Rule: Met 'uri' als uri format, worden alle API links als absolute urls geformatteerd
+
+    Scenario: uri is opgegeven als Accept-Uri-Format header waarde
+        Als GET "/ingeschrevenpersonen?burgerservicenummer=999993847" wordt aangeroepen met HTTP header "Accept-Uri-format"="uri"
+        Dan bevat _links in de response minimaal de volgende links
+        | naam  | href                                                                                                                        | templated |
+        | self  | https://www.haalcentraal.nl/haalcentraal/api/brp/ingeschrevenpersonen?burgerservicenummer=999993847                         |           |
+        | adres | https://api.bag.acceptatie.kadaster.nl/esd/huidigebevragingen/v1/adressen?adresseerbaarObjectIdentificatie=0518010000781379 |           |
+        En bevat de response de volgende HTTP header
+        | naam               | waarde |
+        | Content-Uri-Format | uri    |
+
+Rule: Met 'uri-template' als uri format, worden bij alle API links de base url vervangen door een [xxx]serverurl expression
+
+    Scenario: uri-template is opgegeven als uri-format header waarde
+        Als GET "/ingeschrevenpersonen?burgerservicenummer=999993847" wordt aangeroepen met HTTP header "Accept-Uri-format"="uri-template"
+        Dan bevat _links in de response minimaal de volgende links
+        | naam         | href                                                                      | templated |
+        | brpserverurl | https://www.haalcentraal.nl/haalcentraal/api/brp                          |           |
```

Comment:
ik begrijp uit dit voorbeeld dat elke gebruikte Uri-template expressie als link wordt opgenomen. Wat is dan het nut van Uri-template voor de gebruiker? 
Ik dacht dat Uri-template juist nuttig is, omdat een gebruiker achter een API gateway kan zitten en dus aangepaste base url's moet gebruiken. Dit kan de gebruiker dan eenvoudig invullen door de expressie te vervangen in de hrefs.
Als de gebruiker graag de base uri wil gebruiken die bij de provider bekend is, kan die toch veel beter "Accept-Uri-format"="uri" gebruiken?

Reply:
Mee eens. Bij het API Gateway scenario zal de lijst externe base-url's bekend zijn in de gateway. Het is dan niet nuttig dit in elke response terug te geven.

Reply:
Ik weet niet of ik dit helemaal kan volgen. Ik ga uit dat met gebruiker wordt bedoeld de client applicatie. Die zal in de praktijk uri of uri-reference als uri format gebruiken, denk ik. Ik verwacht dat API Gateways uri-template gaan gebruiken. Er kunnen in principe meerdere API gateways tussen een client en provider zitten. Dan lijkt het mij niet handig als bij elke API gateway een vertaling plaats vindt. Performance wise is het dan beter dat alleen de API Gateway vertaalt die de oorspronkelijke client request heeft ontvangen.